### PR TITLE
[do not merge] v5 GA audit

### PR DESCRIPTION
**Pull request by demerino**
### Summary of changes
A discussion for additional cleanup.

### Authors
- @demerino

### CODE REVIEW

File: Sources/BraintreeCard/Public/BraintreeCard/BTCard.h [hunk 1]
```diff
@@ -10,6 +10,7 @@ NS_ASSUME_NONNULL_BEGIN
 /**
  A convenience initializer for creating a card tokenization request.
 */
+// Consider removing this and just using the properties. Clarifies how to make cvv only nonces?
```
Comment (scannillo) [inline Sources/BraintreeCard/Public/BraintreeCard/BTCard.h:13]: I like the idea of removing these initializers in favor of using the properties.

Good callout - we've seen a handful of questions on how to make a cvv-only nonce. I think it is worth adding a snippet to the docs on clarifying how to make a cvv-only nonce.

File: Sources/PayPalDataCollector/Public/PayPalDataCollector/PPDataCollector.swift [hunk 2]
```diff
@@ -12,6 +12,7 @@ import Security
      - Parameter pairingID: A pairing ID to associate with this clientMetadataID must be 10-32 chars long or null
      - Returns: A client metadata ID to send as a header
      */
+    // Have we decided on Id or ID?
```
Comment (scannillo) [inline Sources/PayPalDataCollector/Public/PayPalDataCollector/PPDataCollector.swift:15]: Addressed in PR https://github.com/braintree/braintree_ios/pull/601

File: Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalRequest.h [hunk 3]
```diff
@@ -166,6 +166,7 @@ typedef NS_ENUM(NSInteger, BTPayPalRequestUserAction) {
 /**
  Optional: Offers PayPal Credit if the customer qualifies. Defaults to false. Only available with PayPal Checkout and PayPal Billing Agreement.
  */
+// Can we remove this?
```
Comment (scannillo) [inline Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalRequest.h:169]: Does the Pay Later offering intend to completely replace PayPal credit?

File: Sources/BraintreeVenmo/Public/BraintreeVenmo/BTVenmoDriver.h [hunk 4]
```diff
@@ -67,6 +67,7 @@ typedef NS_ENUM(NSInteger, BTVenmoDriverErrorType) {
     On success, you will receive an instance of `BTVenmoAccountNonce`; on failure, an error; on user
     cancellation, you will receive `nil` for both parameters.
 */
+// We should rework this to use a `BTVenmoRequest` object and consolidate to a single authorization method. There are projects coming to move to Payment Context and add more customization to the experience
```
Comment (scannillo) [inline Sources/BraintreeVenmo/Public/BraintreeVenmo/BTVenmoDriver.h:70]: This is a great idea! 

Addressed in PR https://github.com/braintree/braintree_ios/pull/602

File: Sources/BraintreePaymentFlow/BTPaymentFlowDriver.m [hunk 5]
```diff
@@ -74,6 +74,7 @@ - (void)performSwitchRequest:(NSURL *)appSwitchURL {
 
 - (void)informDelegatePresentingViewControllerRequestPresent:(NSURL *)appSwitchURL {
     if ([self.viewControllerPresentingDelegate respondsToSelector:@selector(paymentDriver:requestsPresentationOfViewController:)]) {
+        // Why not ASWebAuthenticationSession here?
```
Comment (scannillo) [inline Sources/BraintreePaymentFlow/BTPaymentFlowDriver.m:77]: After conversation in stand-up, we decided we will not make this change to BTPaymentFlowDriver at the moment. It requires the user to agree to an alert view, which has the `braintreegateway.com` URL. We tabled the work for it but left a working branch up at `aswebauth-payment-flow-driver`.

File: Sources/BraintreeCore/Public/BraintreeCore/BTConfiguration.h [hunk 6]
```diff
@@ -6,6 +6,7 @@ NS_ASSUME_NONNULL_BEGIN
 /**
  Contains information specific to a merchant's Braintree integration
  */
+// Would it make sense to collapse all of the Configuration+ categories into a single class here?
```
Comment (sestevens) [inline Sources/BraintreeCore/Public/BraintreeCore/BTConfiguration.h:9]: We looked into this, and we actually like how the configuration categories are organized now. We think having all the properties in one class might be too much.

We did open a PR to add an `environment` property to `BTConfiguration`, though: #603

File: Sources/BraintreeCard/Public/BraintreeCard/BTCard.h (hunk 1 above)
Comment (sestevens) [reply Sources/BraintreeCard/Public/BraintreeCard/BTCard.h:13]: Update in #604

File: Sources/BraintreeCard/Public/BraintreeCard/BTCard.h [hunk 7]
```diff
@@ -18,6 +19,7 @@ NS_ASSUME_NONNULL_BEGIN
 /**
 Designated initializer.
  */
+// Consider removing, the dictionary style of initialization isn't used elsewhere?
```
Comment (sestevens) [inline Sources/BraintreeCard/Public/BraintreeCard/BTCard.h:22]: Removed in #604

File: Sources/BraintreeCard/BTCardClient.m [hunk 8]
```diff
@@ -244,6 +244,7 @@ - (BOOL)isGraphQLEnabledForCardTokenization:(BTConfiguration *)configuration {
     return graphQLFeatures && [graphQLFeatures containsObject:BTCardClientGraphQLTokenizeFeature];
 }
 
+// Remove all of this data collection for each tokenization, it has been removed from the gateway
```
Comment (sestevens) [inline Sources/BraintreeCard/BTCardClient.m:247]: Removed in #605

File: Sources/BraintreeCore/Public/BraintreeCore/BTAppSwitch.h [hunk 9]
```diff
@@ -67,16 +67,19 @@ NS_ASSUME_NONNULL_BEGIN
 /**
  Notification that an app switch will occur
  */
+// Remove in favor of BTAppContextWillSwitchNotification?
```
Comment (sestevens) [inline Sources/BraintreeCore/Public/BraintreeCore/BTAppSwitch.h:70]: These NSNotifications and `BTAppSwitchDelegate` were removed in #606

File: Sources/BraintreePayPal/BTPayPalDriver.m [hunk 10]
```diff
@@ -321,9 +321,12 @@ - (void)performSwitchRequest:(NSURL *)appSwitchURL paymentType:(BTPayPalPaymentT
 
     NSURLComponents *urlComponents = [NSURLComponents componentsWithURL:appSwitchURL resolvingAgainstBaseURL:NO];
 
+    // Verify what bt_int_type does
     NSString *queryForAuthSession = [urlComponents.query stringByAppendingString:@"&bt_int_type=2"];
     urlComponents.query = queryForAuthSession;
 
+    // Verify this URL matches v4 URLs
```
Comment (scannillo) [inline Sources/BraintreePayPal/BTPayPalDriver.m:328]: ✅ I compared the URLs constructed for v4 vs v5, and they are structured the same with the same query params.

File: Sources/BraintreePayPal/BTPayPalDriver.m [hunk 11]
```diff
@@ -321,9 +321,12 @@ - (void)performSwitchRequest:(NSURL *)appSwitchURL paymentType:(BTPayPalPaymentT
...
+    // Verify that RDA data is being sent correctly
```
Comment (scannillo) [inline Sources/BraintreePayPal/BTPayPalDriver.m:329]: We should be good on this front. In both flows - we're sending the RDA data at the same time: after fetching the URL from `paypal_hermes`, before sending analytics data, and before continuing on with the flow via `handlePayPalRequest()`.

In v4, we sent RDA data [here](https://github.com/braintree/braintree_ios/blob/6da9607d732e45df512bc82a5a5c7c2461d9d2c2/BraintreePayPal/PayPalOneTouch/Models/PPOTSwitchRequest.m#L44) and were [calling](https://github.com/braintree/braintree_ios/blob/6da9607d732e45df512bc82a5a5c7c2461d9d2c2/BraintreePayPal/BTPayPalRequestFactory.m#L15) `tokenFromApprovalURL` function to extract the `pairingID`. 

In v5, we both extract the `pairingID` and send the RDA data entirely in `BTPayPalDriver` [here](https://github.com/braintree/braintree_ios/blob/e653dba9e975272112547d3d72a6ea31cc99d72a/Sources/BraintreePayPal/BTPayPalDriver.m#L256-L258).

File: Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalRequest.h (hunk 3 above)
Comment (sestevens) [reply Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalRequest.h:169]: @demerino The doc string for `offerCredit` says that it applies to both checkout and billing agreement flows, whereas `offerPayLater` says that it only applies to checkout. Should we keep `offerCredit` just for the billing agreement flow?

File: Sources/BraintreeDataCollector/Public/BraintreeDataCollector/BTDataCollector.h [hunk 12]
```diff
@@ -72,6 +73,7 @@ extern NSString * const BTDataCollectorKountErrorDomain;
 /**
  Provides status updates from a BTDataCollector instance. At this time, updates will only be sent for card fraud data (from Kount).
 */
+// Do the other SDKs have this feature? Consider removing.
```
Comment (scannillo) [inline Sources/BraintreeDataCollector/Public/BraintreeDataCollector/BTDataCollector.h:76]: Removed in PR #613

File: Sources/BraintreeDataCollector/Public/BraintreeDataCollector/BTDataCollector.h [hunk 13]
```diff
@@ -54,6 +54,7 @@ extern NSString * const BTDataCollectorKountErrorDomain;
 
  @param completion A completion block callback that returns a deviceData string that should be passed in to server-side calls, such as `Transaction.sale` This JSON serialized string contains the merchant ID and session ID.
 */
+// consider removing, kount will be collected by the other method if the merchant is setup for it
```
Comment (scannillo) [inline Sources/BraintreeDataCollector/Public/BraintreeDataCollector/BTDataCollector.h:57]: Removed in PR #613

File: Sources/BraintreePayPal/BTPayPalDriver.m [hunk 14]
```diff
@@ -321,9 +321,12 @@ - (void)performSwitchRequest:(NSURL *)appSwitchURL paymentType:(BTPayPalPaymentT
 
     NSURLComponents *urlComponents = [NSURLComponents componentsWithURL:appSwitchURL resolvingAgainstBaseURL:NO];
 
+    // Verify what bt_int_type does
```
Comment (scannillo) [inline Sources/BraintreePayPal/BTPayPalDriver.m:324]: Removed in PR #614

File: Sources/BraintreeCore/Public/BraintreeCore/BTAppSwitch.h [hunk 15]
```diff
@@ -99,6 +102,7 @@ extern NSString * const BTAppContextDidReturnNotification;
 /**
  Specifies the destination of an app switch
 */
+// I think all app swtiching is now either/or - are there flows where this information is ambiguous?
```
Comment (scannillo) [inline Sources/BraintreeCore/Public/BraintreeCore/BTAppSwitch.h:105]: Removed in #606

File: Sources/BraintreeCore/Public/BraintreeCore/BTAppSwitch.h [hunk 16]
```diff
@@ -67,16 +67,19 @@ NS_ASSUME_NONNULL_BEGIN
...
 /**
  Notification that an app switch did occur
  */
+// Remove in favor of BTAppContextDidReturnNotification?
 extern NSString * const BTAppSwitchDidSwitchNotification;
 
 /**
  Notification that an app switch will process payment information
  */
+// Can we remove this?
```
Comment (scannillo) [inline Sources/BraintreeCore/Public/BraintreeCore/BTAppSwitch.h:82]: Removed in https://github.com/braintree/braintree_ios/pull/606

File: Sources/BraintreeCore/Public/BraintreeCore/BTAppSwitch.h [hunk 17]
```diff
@@ -67,16 +67,19 @@ NS_ASSUME_NONNULL_BEGIN
 /**
  Notification that an app switch will occur
  */
+// Remove in favor of BTAppContextWillSwitchNotification?
 extern NSString * const BTAppSwitchWillSwitchNotification;
 
 /**
  Notification that an app switch did occur
  */
+// Remove in favor of BTAppContextDidReturnNotification?
```
Comment (scannillo) [inline Sources/BraintreeCore/Public/BraintreeCore/BTAppSwitch.h:76]: Removed in https://github.com/braintree/braintree_ios/pull/606

File: Sources/BraintreeCore/Public/BraintreeCore/BTLogger.h [hunk 18]
```diff
@@ -3,6 +3,7 @@
 /**
  Braintree SDK Logging Levels
 */
+// Usage of this logger seems sparse
```
Comment (scannillo) [inline Sources/BraintreeCore/Public/BraintreeCore/BTLogger.h:6]: I think there are places where having logs is helpful to merchants (for example:[ in BTAPIClient when their auth credentials are invalid](https://github.com/braintree/braintree_ios/blob/a234cdd0e070590b150c09f7d310f27198d6237e/Sources/BraintreeCore/BTAPIClient.m#L71)).

I do think we could consolidate some of these log levels, though. I think this doesn't need to be done prior to the v5 GA release.

File: Sources/BraintreeCore/Public/BraintreeCore/BTPostalAddress.h [hunk 19]
```diff
@@ -10,6 +10,7 @@ NS_ASSUME_NONNULL_BEGIN
 /**
  Optional. Recipient name for shipping address.
 */
+// Maybe `fullName` (see graphql) with a note: Full name, (eg. recipient when shipping).
```
Comment (scannillo) [inline Sources/BraintreeCore/Public/BraintreeCore/BTPostalAddress.h:13]: I assume if we change to `fullName` we want to make this change across all SDKs (which still use `recipientName`).

Also, the [GraphQL docs ](https://graphql.braintreepayments.com/reference/#object--address) also don't use `PostalAddress` but rather `Address`. Is this also a change we want to make?

Comment (scannillo) [reply Sources/BraintreeCore/Public/BraintreeCore/BTPostalAddress.h:13]: @demerino

File: Sources/BraintreeCore/Public/BraintreeCore/BTLogger.h (hunk 18 above)
Comment (sestevens) [reply Sources/BraintreeCore/Public/BraintreeCore/BTLogger.h:6]: +1

File: Sources/BraintreeCore/Public/BraintreeCore/BTPostalAddress.h (hunk 19 above)
Comment (sestevens) [reply Sources/BraintreeCore/Public/BraintreeCore/BTPostalAddress.h:13]: I think we should punt on this, unless we think this is particularly confusing or problematic for merchants. There are probably quite a few fields where the SDK names don't match GraphQL, but I don't think that should hold up the release.

Maybe consistency with GraphQL can be an objective of the v6 release?

Comment (scannillo) [reply Sources/BraintreeCore/Public/BraintreeCore/BTPostalAddress.h:13]: Confirmed: punting

**Comment by scannillo** [comment]
We've addressed all these comments. Thank you for the thorough feedback @demerino! If you don't have any follow-ups, we can close this PR.

File: Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalRequest.h (hunk 3 above)
Comment (sestevens) [reply Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalRequest.h:169]: Verified with David that we can remove `offerCredit`.

Comment (sestevens) [reply Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalRequest.h:169]: Addressed in #618

**Pull request closed without merging**